`quarkus-maven-plugin`: dependency information is stored in the maven context and not properly cleaned up

### Describe the bug

It seems that the `quarkus-maven-plugin` stores some information in the maven context. Those information is not properly cleaned up and can influence other plugins.

The problem is observable with the `maven-enforcer-plugin`, especially the builtin [`BanDependencyManagementScope`-rule (`github.com`)](enforcer-rules/src/main/java/org/apache/maven/enforcer/rules/BanDependencyManagementScope.java).

To illustrate, we can look at [this multimodule project (`github.com`)](https://github.com/turing85/quarkus-zeromq).

---

What works: a project has the following overall structure

```
.
├── bom
├── commons
├── service-one
└── service-two
```

(this reflects the repo above on branch `main`)

All submodules load `bom` in their dependency management (except for `bom` itself of course), and all submodules use the root module as their parent.

This works fine, we can `./mvnw validate` and `./mvnw package` without problem.

---

What does not work: a project has the following overall structure

```
.
├── bom
├── commons
├── quarkus-service-parent
├── service-one
└── service-two
```

(this reflects the repo above on branch `quarkus-service-parent`)

Modules `commons` and `quarkus-service-parent` now load `bom` in their dependency management, and continue to use the root module as their parent, while modules `service-one` and `service-two` use `quarkus-service-parent` as their parent (and have no `dependencyManagement` section).

Running `./mvnw validate` works.

Runningn `./mvnw package` results in the follwing build error:

```
...
[INFO] --- enforcer:3.6.1:enforce (default) @ quarkus-zeromq-service-two ---
[INFO] Rule 0: org.codehaus.mojo.extraenforcer.dependencies.BanCircularDependencies passed
[INFO] Rule 2: org.apache.maven.enforcer.rules.BanDuplicatePomDependencyVersions passed
[INFO] Rule 3: org.apache.maven.enforcer.rules.dependency.BanDynamicVersions passed
[INFO] Rule 4: org.apache.maven.enforcer.rules.BannedPlugins passed
[INFO] Rule 5: org.apache.maven.enforcer.rules.dependency.DependencyConvergence passed
[INFO] Rule 8: org.apache.maven.enforcer.rules.RequirePluginVersions passed
[INFO] Rule 9: org.apache.maven.enforcer.rules.dependency.RequireReleaseDeps passed
[INFO] Rule 10: org.apache.maven.enforcer.rules.RequireSameVersions passed
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for quarkus-zeromq :: parent 999-SNAPSHOT:
[INFO] 
[INFO] quarkus-zeromq :: parent ........................... SUCCESS [  1.110 s]
[INFO] quarkus-zeromq :: bom .............................. SUCCESS [  0.050 s]
[INFO] quarkus-zeromq :: commons .......................... SUCCESS [  3.829 s]
[INFO] quarkus-zeromq :: quarkus service parent ........... SUCCESS [  0.730 s]
[INFO] quarkus-zeromq :: service one ...................... SUCCESS [ 10.765 s]
[INFO] quarkus-zeromq :: service two ...................... FAILURE [  0.407 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  18.541 s
[INFO] Finished at: 2025-08-01T19:09:09+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.6.1:enforce (default) on project quarkus-zeromq-service-two: 
[ERROR] Rule 1: org.apache.maven.enforcer.rules.BanDependencyManagementScope failed with message:
[ERROR] Scope other than 'import' is not allowed in 'dependencyManagement'
[ERROR] Banned scope 'provided' used on dependency 'io.quarkus:quarkus-observability-devservices-lgtm:jar' @ io.quarkus.platform:quarkus-bom:3.25.0, /eXchange/.m2/repository/io/quarkus/platform/quarkus-bom/3.25.0/quarkus-bom-3.25.0.pom, line 4851, column 19
[ERROR] Banned scope 'provided' used on dependency 'org.jboss.logging:jboss-logging-annotations:jar' @ io.quarkus.platform:quarkus-bom:3.25.0, /eXchange/.m2/repository/io/quarkus/platform/quarkus-bom/3.25.0/quarkus-bom-3.25.0.pom, line 9563, column 19
... (lots and lots of those errors)
[ERROR] Banned scope 'runtime' used on dependency 'org.mvnpm:yargs:jar' @ io.quarkus.platform:quarkus-bom:3.25.0, /eXchange/.m2/repository/io/quarkus/platform/quarkus-bom/3.25.0/quarkus-bom-3.25.0.pom, line 10705, column 19
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :quarkus-zeromq-service-two
```

Building with the maven daemon (`mvnd package`) succeeds.

### Expected behavior

Enforcer rules are applied, and project builds successfully.

### Actual behavior

When `./mvnw package` is used, the build fails with the above error.

### How to Reproduce?

Reproducer:

1. clone `https://github.com/turing85/quarkus-zeromq`:

```bash
git clone https://github.com/turing85/quarkus-zeromq.git && cd quarkus-zeromy
```
2. Build the project, observe that the enforcer plugin is executed, and the build succeeds:
```bash
./mvnw package
```
3. Switch to branch `quarkus-service-parent`:
```bash
git switch quarkus-service-parent
```
4. Run the validation phase, observe that the enforcer plugin is executed, and validation succeeds:
```bash
./mvnw validate
```
5. Build the project, observe the above error:
```bash
./mvnw package
```
6. Buid the project with the maven daemon, observe that the enoforcer plugin is executed, and the project builds successfully:
```bash
mvnd package
```

### Output of `uname -a` or `ver`

```
Linux xxx 6.8.0-71-generic #71-Ubuntu SMP PREEMPT_DYNAMIC Tue Jul 22 16:52:38 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux
```

### Output of `java -version`

```
openjdk version "21.0.7" 2025-04-15 LTS
OpenJDK Runtime Environment Temurin-21.0.7+6 (build 21.0.7+6-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.7+6 (build 21.0.7+6-LTS, mixed mode, sharing)
```

### Quarkus version or git rev

since (at least) 3.24.5 

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

maven:
```
Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: /home/marco/.m2/wrapper/dists/apache-maven-3.9.9/3477a4f1
Java version: 21.0.7, vendor: Eclipse Adoptium, runtime: /opt/java/mandrel/23.1.7.0-java21
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.8.0-71-generic", arch: "amd64", family: "unix"
```

maven daemon:
```
Apache Maven Daemon (mvnd) 1.0.2 linux-amd64 native client (cccc1ec8a5b741c62b29f9fb04d37b1678a029bb)
Terminal: org.jline.terminal.impl.PosixSysTerminal with pty org.jline.terminal.impl.jni.linux.LinuxNativePty
Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)
Maven home: /opt/mvnd/1.0.2/mvn
Java version: 21.0.7, vendor: Eclipse Adoptium, runtime: /opt/java/mandrel/23.1.7.0-java21
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.8.0-71-generic", arch: "amd64", family: "unix"
```


### Additional information

_No response_

What happens here?
https://github.com/quarkusio/quarkus/blob/d18f6d2448fd581612cb0087811ce42b8dc98216/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java#L112-L124

As enforcer fails exactly due this: "original model" (so not the effective, but the model on disk) contains depMgt with scopes, but message clearly says they come from Quarkus BOM (and not "original" user edited POM):

```
[ERROR] Banned scope 'provided' used on dependency 'io.quarkus:quarkus-observability-devservices-lgtm:jar' @ io.quarkus.platform:quarkus-bom:3.25.0, /eXchange/.m2/repository/io/quarkus/platform/quarkus-bom/3.25.0/quarkus-bom-3.25.0.pom, line 4851, column 19
```

--- a/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/QuarkusBootstrapProvider.java
@@ -110,8 +110,11 @@ public class QuarkusBootstrapProvider implements Closeable {
      * @return raw POM
      */
     private static Model getRawModel(MavenProject mp) {
-        final Model model = mp.getOriginalModel();
+        Model model = mp.getOriginalModel();
         if (model.getDependencyManagement() == null) {
+            // clone the model to not modify the original model associated with the project,
+            // otherwise, the enforcer plugin may fail, for example
+            model = model.clone();
             // this could be the flatten plugin removing the dependencyManagement
             // in which case we set the effective dependency management to not lose the platform info
             model.setDependencyManagement(mp.getDependencyManagement());

